frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -0,0 +1,156 @@
+} from "./EmbedFrame.styled";
+import "./EmbedFrame.css";
+
+interface OwnProps {
+  className?: string;
+  name?: string;
+  description?: string;
+  dashboard?: Dashboard;
+  actionButtons?: JSX.Element[];
+  parameters?: Parameter[];
+  parameterValues?: Record<ParameterId, ParameterValueOrArray>;
+  setParameterValue?: (parameterId: ParameterId, value: any) => void;
+  children: React.ReactNode;
+}
+
+interface StateProps {
+  hasEmbedBranding: boolean;
+}
+
+type Props = OwnProps &
+  StateProps & {
+    location: Location;
+  };
+
+interface HashOptions {
+  bordered?: boolean;
+  titled?: boolean;
+  theme?: string;
+  hide_parameters?: string;
+  hide_download_button?: boolean;
+}
+
+function mapStateToProps(state: State) {
+  return {
+    hasEmbedBranding: !getSetting(state, "hide-embed-branding?"),
+  };
+}
+
+function EmbedFrame({
+  className,
+  children,
+  name,
+  description,
+  dashboard,
+  actionButtons,
+  location,
+  hasEmbedBranding,
+  parameters,
+  parameterValues,
+  setParameterValue,
+}: Props) {
+  const [hasInnerScroll, setInnerScroll] = useState(true);
+
+  useOnMount(() => {
+    initializeIframeResizer(() => setInnerScroll(false));
+  });
+
+  const {
+    bordered = IFRAMED,
+    titled = true,
+    theme,
+    hide_parameters,
+    hide_download_button,
+  } = parseHashOptions(location.hash) as HashOptions;
+
+  const showFooter =
+    hasEmbedBranding || (!hide_download_button && actionButtons);
+
+  const finalName = titled ? name : null;
+
+  const hasParameters =
+    Array.isArray(parameters) &&
+    parameters.length > 0 &&
+    typeof setParameterValue === "function";
+
+  const hasHeader = Boolean(finalName || hasParameters);
+
+  return (
+    <Root
+      hasScroll={hasInnerScroll}
+      isBordered={bordered}
+      className={cx("EmbedFrame", className, {
+        [`Theme--${theme}`]: !!theme,
+      })}
+    >
+      <ContentContainer hasScroll={hasInnerScroll}>
+        {hasHeader && (
+          <Header className="EmbedFrame-header">
+            {finalName && (
+              <TitleAndDescription
+                title={finalName}
+                description={description}
+                className="my2"
+              />
+            )}
+            {hasParameters && (
+              <div className="flex">
+                <SyncedParametersList
+                  className="mt1"
+                  dashboard={dashboard}
+                  parameters={getValuePopulatedParameters(
+                    parameters,
+                    parameterValues,
+                  )}
+                  setParameterValue={setParameterValue}
+                  hideParameters={hide_parameters}
+                />
+              </div>
+            )}
+          </Header>
+        )}
+        <Body>{children}</Body>
+      </ContentContainer>
+      {showFooter && (
+        <Footer className="EmbedFrame-footer">
+          {hasEmbedBranding && <LogoBadge dark={theme === "night"} />}
```

Comment:
Wait! was the previous version causing a bug the whole time? It just passes `theme` to `dark` without checking if its value is `night`. And how to you know this needs to be `night`?

Reply:
> And how to you know this needs to be `night`?

From the [docs](https://www.metabase.com/docs/latest/questions/sharing/public-links#appearance-parameters)

Reply:
@WiNloSt I've extracted this change into a separate PR #27796, so we don't mix refactoring with bug fixing